src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPoolIterator2.cs

```diff
@@ -90,6 +90,49 @@ public int Take()
             }
         }
 
+        public void Skip(int bytesToSkip)
```

Comment:
Please add unit tests for this new method. Have a look at `Microsoft.AspNet.Server.KestrelTests.MemoryPoolIterator2Tests.PeekLongAtBlockBoundary` if you need an idea as to how to test close to block boundaries.